src/Tools/vmOpsTool/VMWareImpl.java

```diff
@@ -608,7 +608,8 @@ private ManagedObjectReference findSnapshotInTree(List<VirtualMachineSnapshotTre
 
             if (vmSnapshot.getName().equalsIgnoreCase(snapshotName)) {
                 System.out.println(String.format("Found snapshot [ %s ] for virtual machine.", snapshotName));
-                return vmSnapshot.getSnapshot();
+                snapshotMor =  vmSnapshot.getSnapshot();

```

Comment:
You need to break for loop before doing another iteration for next child if snapshotMor is not null. So breaking here is not correct thing it will fix

For example snapshot PS1 has immediate childs CS1, CS2, and CS3. Where as CS2 has grand child GCS 1, now if we are searching for GCS1 this won't work.